src/test/groovy/com/synacy/gradprogram/spock/exercise/DeliveryServiceSpec.groovy

```diff
@@ -0,0 +1,41 @@
+package com.synacy.gradprogram.spock.exercise
+
+import spock.lang.Specification
+
+class DeliveryServiceSpec extends Specification {
+    DeliveryRequestRepository deliveryRequestRepository
+    DateUtils dateUtils
+
+    DeliveryService deliveryService
+
+    void setup() {
+        deliveryRequestRepository = Mock()
+        dateUtils = Mock()
+        deliveryService = new DeliveryService(dateUtils, deliveryRequestRepository)
+    }
+
+    def "createDeliveryRequest should save delivery request with the valid order details"() {
+        given:
+        def currentDate = new Date()
+        dateUtils.currentDate >> currentDate
+
+        def order = new Order()
+        order.totalCost = expectedTotalCost
+
+        when:
+        deliveryService.createDelivery(order)
+
+        then:
+        1 * deliveryRequestRepository.saveDeliveryRequest(_ as DeliveryRequest) >> { DeliveryRequest passedRequest ->
+            assert order.id == passedRequest.orderId
+            assert currentDate == passedRequest.deliveryDate
+            assert expectedCourier == passedRequest.courier
+        }
+
+        where:
+        expectedCourier | expectedTotalCost | desc
```

Comment:
We add the prefix `expected` to variable names that we use in the assertion. For the `expectedTotalCost`, since we don't use it there, we can just name it as `totalCost` instead :+1: .

The `desc` isn't being used, it would be better to incorporate it in the test method name.